src/CommunityToolkit.Maui.Core/Essentials/FolderPicker/FolderPickerImplementation.android.cs

```diff
@@ -12,20 +13,24 @@ public sealed partial class FolderPickerImplementation : IFolderPicker
 {
 	async Task<Folder> InternalPickAsync(string initialPath, CancellationToken cancellationToken)
 	{
-		var status = await Permissions.RequestAsync<Permissions.StorageRead>().WaitAsync(cancellationToken).ConfigureAwait(false);
-		if (status is not PermissionStatus.Granted)
+		const string baseExternalStorageUrl = "content://com.android.externalstorage.documents/document/primary%3A";
+		Folder? folder = null;
+
+		if (!OperatingSystem.IsAndroidVersionAtLeast(33))
 		{
-			throw new PermissionException("Storage permission is not granted.");
+			var statusRead = await Permissions.RequestAsync<Permissions.StorageRead>().WaitAsync(cancellationToken).ConfigureAwait(false);
+			if (statusRead is not PermissionStatus.Granted)
+			{
+				throw new PermissionException("Storage permission is not granted.");
+			}
 		}
 
-		Folder? folder = null;
-		const string baseUrl = "content://com.android.externalstorage.documents/document/primary%3A";
 		if (Android.OS.Environment.ExternalStorageDirectory is not null)
 		{
 			initialPath = initialPath.Replace(Android.OS.Environment.ExternalStorageDirectory.AbsolutePath, string.Empty, StringComparison.InvariantCulture);
 		}
 
-		var initialFolderUri = AndroidUri.Parse(baseUrl + HttpUtility.UrlEncode(initialPath));
+		var initialFolderUri = AndroidUri.Parse(baseExternalStorageUrl + HttpUtility.UrlEncode(initialPath));
 
 		var intent = new Intent(Intent.ActionOpenDocumentTree);
 		intent.PutExtra(DocumentsContract.ExtraInitialUri, initialFolderUri);
```

Comment:
`DocumentsContract.ExtraInitialUri` is only supported in Android API 26+. We support Android 21+ because .NET MAUI supports Android 21+. 

Could we add a branch to support Android API 21-25?

I know this bug isn't introduced in this PR, but any chance that it's a quick/easy fix we could squeeze in here?